src/internal_modules/roc_pipeline/sender_endpoint.cpp

```diff
@@ -99,10 +99,16 @@ SenderEndpoint::SenderEndpoint(address::Protocol proto,
     }
 
     composer_ = composer;
+    if (!composer_) {
+        return;
+    }
+
+    writer_.reset(new (arena) packet::Shipper(dest_address, *composer_, dest_writer),
+                  arena);
```

Comment:
Instead of `core::ScopedPtr<packet::IWriter>`, let's use `core::Optional<packet::Shipper>` to avoid unneeded allocation.